src/borg/archive.py

```diff
@@ -1034,28 +1034,35 @@ def identify_key(self, repository):
         return key_factory(repository, cdata)
 
     def verify_data(self):
+        def chunker(it, size):
+            while True:
+                result = []
+                try:
+                    for i in range(size):
+                        result.append(next(it))
+                except StopIteration:
+                    yield result
+                    raise StopIteration
+                else:
+                    yield result
```

Comment:
@RonnyPfannschmidt see the next commit in this PR, with fixed iterators it works now. :)